doc/faq.rst

```diff
@@ -330,3 +330,21 @@ However, a global random state is prone to modification by other code during
 execution. Thus, the only way to ensure replicability is to pass ``RandomState``
 instances everywhere and ensure that both estimators and cross-validation
 splitters have their ``random_state`` parameter set.
+
+Why do categorical variables need preprocessing in scikit-learn, compared to other tools?
+--------------------------------------------------------------------------------
+
+Most of scikit-learn assumes data is in NumPy arrays or SciPy sparse matrices of
+a single numeric dtype. These do not directly explicitly categorical variables at
+present. See the :class:`preprocessing.CategoricalEncoder` for a tool to encode
+categorical variables, say within a pandas.DataFrame, as numeric data.
+See :ref:`sphx_glr_auto_examples_hetero_feature_union.py` for an example of
+working with heterogeneous (e.g. categorical and numeric) data.
+
+Why does Scikit-learn not directly work with, for example, pandas.DataFrame?
+
+The homogeneous NumPy and SciPy data objects currently expected are most
+efficient to process for most operations. Extensive work would also be needed
+to support Pandas categorical types. Restricting input to homogeneous
+types therefore reduces maintenance cost and encourages usage of efficient
+data structures.
```

Comment:
You're sure you don't want to finish the FAQ connecting it clearly with how scikit-learn handles the original problem (categorical variables)? I was quite happy about my ending line "Facilitating explicit preprocessing is a much more
cost-effective solution."

Reply:
Sorry, but I don't think that sentence means anything.

Reply:
Just in case, I was thinking in the terms amueller used in his comment above 

"Some other tools do IMPLICIT CONVERSIONS, scikit-learn requires EXPLICIT CONVERSIONS
[...]
That [categorical values] have to be dealt with EXPLICITLY is quite surprising for people coming from R, which seems to be a substantial population (I don't think I've ever given a tutorial where people did not ask about the requirements of scikit-learn to EXPLICITLY treat missing values and categorical variables)."

Granted, I said "explicit preprocessing" instead of "facilitating explicit conversion".
---

If it still is wrong and you're happy anyway ok.